dev-python/genshi/genshi-0.7.3-r1.ebuild

```diff
@@ -0,0 +1,39 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} pypy3 )
+
+inherit distutils-r1
+
+DESCRIPTION="Python toolkit for stream-based generation of output for the web"
+HOMEPAGE="http://genshi.edgewall.org/ https://pypi.org/project/Genshi/"
+SRC_URI="mirror://pypi/G/${PN^}/${P^}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux ~x64-macos ~x86-macos"
+IUSE="doc examples test"
+RESTRICT="!test? ( test )"
+
+BDEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
+DEPEND="${BDEPEND}"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/${P^}"
+
+python_test() {
+	esetup.py test || die
+}
```

Comment:
Rather than providing an implementation of `python_test`, you should use. 
```
distutils_enable_tests setup.py
```
You can remove `test` from `IUSE` and `RESTRICT="!test? ( test )"` since those are handled from the eclass with that

Reply:
done